### PR TITLE
HTBHF-2199 Make a call to GA on successful and failed postcode lookup for audit purposes.

### PR DESCRIPTION
**Pull request by mikeSmith08**

### CODE REVIEW

File: src/web/routes/application/postcode/postcode.js [hunk 1]
```diff
@@ -8,6 +8,25 @@ const REQUEST_TIMEOUT = 5000
...
 const standardisePostcode = (postcode) => postcode.toUpperCase().replace(/\s/g, '')
 
+const buildGoogleAnalyticsFullUri = (gaBaseUrl, trackingId, result, ipAddress, numberOfResults, sessionId) =>
+  `${gaBaseUrl}?v=1&tid=${trackingId}&t=event&cid=${sessionId}&ec=AddressLookup&ea=${result}&el=${ipAddress}&ev=${numberOfResults}`
+
+const auditSuccessfulPostcodeLookup = async (config, req, numberOfResults) => auditPostcodeLookup(config, req, 'SuccessfulLookup', numberOfResults)
+
+const auditFailedPostcodeLookup = async (config, req) => auditPostcodeLookup(config, req, 'FailedLookup', 0)
+
+const auditPostcodeLookup = async (config, req, result, numberOfResults) => {
```
Comment (YetAnotherMatt) [inline src/web/routes/application/postcode/postcode.js]: Maybe 'result' could be renamed 'outcome' to make it different to 'numberOfResults'

File: manifest.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ applications:
     CLAIMANT_SERVICE_URL: http://htbhf-claimant-service((space-suffix)).apps.internal:8080
     NODE_ENV: production
     APP_VERSION: ((app-version))
+    GOOGLE_ANALYTICS_URI: https://www.google-analytics.com/collect
```
Comment (gethyn1) [inline manifest.yml]: Is this the only Google Analytics URI we could use? If we decided to track events, would that be the same URI?

File: src/web/routes/application/postcode/postcode.js [hunk 3]
```diff
@@ -8,6 +8,25 @@ const REQUEST_TIMEOUT = 5000
 
 const standardisePostcode = (postcode) => postcode.toUpperCase().replace(/\s/g, '')
 
+const buildGoogleAnalyticsFullUri = (gaBaseUrl, trackingId, result, ipAddress, numberOfResults, sessionId) =>
+  `${gaBaseUrl}?v=1&tid=${trackingId}&t=event&cid=${sessionId}&ec=AddressLookup&ea=${result}&el=${ipAddress}&ev=${numberOfResults}`
+
+const auditSuccessfulPostcodeLookup = async (config, req, numberOfResults) => auditPostcodeLookup(config, req, 'SuccessfulLookup', numberOfResults)
```
Comment (gethyn1) [inline src/web/routes/application/postcode/postcode.js]: Do any of these functions need to be `async` as the code doesn't `await` the result?

Comment (mikeSmith08) [reply src/web/routes/application/postcode/postcode.js]: Looks like you're right, have removed them and all works well.

File: src/web/routes/application/postcode/postcode.js (hunk 1 above)
Comment (mikeSmith08) [reply src/web/routes/application/postcode/postcode.js]: Done

**Pull request merged**